src/source/Ice/IceUtils.c

```diff
@@ -183,7 +183,7 @@ STATUS populateIpFromString(PKvsIpAddress pKvsIpAddress, PCHAR pBuff)
 
     curr = pBuff;
     tail = pBuff + STRLEN(pBuff);
-    while ((next = STRNCHR(curr, tail - curr, '.')) != NULL) {
+    while ((next = STRNCHR(curr, tail - curr, '.')) != NULL && octet < 3) {
```

Comment:
Why octet < 3? Put a constant or comment.

Reply:
added new comment "first 3 octet should always end with a '.', the last octet may end with ' ' or '\0""